src/ipc/handler.c

```diff
@@ -1064,6 +1064,64 @@ static int ipc_glb_tplg_message(uint32_t header)
 	}
 }
 
+static int ipc_glb_large_message(uint32_t header)
+{
+	struct sof_ipc_comp_reply reply;
+	struct sof_ipc_large_hdr *hdr;
+	struct sof_ipc_comp *lg = (struct sof_ipc_comp *)_ipc->large_data;
+	uint32_t type;
+	uint32_t offset;
+
+	reply.rhdr.hdr.cmd = header;
+	reply.rhdr.hdr.size = sizeof(reply);
+	reply.rhdr.error = 0;
+	reply.offset = 0;
+
+	hdr = (struct sof_ipc_large_hdr *)_ipc->comp_data;
+
+	/* TODO: what is the max large data size? */
+	if (hdr->id == 0) {
+		_ipc->large_data = rzalloc(RZONE_RUNTIME, SOF_MEM_CAPS_RAM,
```

Comment:
I believe that we agreed to allocate memory in the target component instance, not in the IPC handler layer. In that case no max large data size need to be defined upfront globally.

Reply:
sorry wasn't part of the discussion.. was the idea that fw will ask the component to allocate memory, then get this allocated area back and write the ipc there? at which point is the old parameter memory destroyed, I guess when reserving new mem?

Reply:
@juimonen yep, component does the allocation since it has knowledge about use case (and can reject invalid requests)

Reply:
For EQ the component would need to allocate for worst-case always. If there are many instances it creates a large overhead. Since there can be in practice very light EQs (e.g. a good microphone) or a large EQs (or a large set of preset EQs) it would be useful to be able to specify the worst-case per instance.